presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -5301,6 +5301,113 @@ public void testUnsupportedCsvTable()
+                ".*Unpartitioned materialized view is not supported.");
+        assertQueryFails(
+                "CREATE MATERIALIZED VIEW test_customer_mv_no_direct_partition_mapped WITH (partitioned_by = ARRAY['nationkey']) AS SELECT name, CAST(nationkey AS BIGINT) AS nationkey FROM test_customer_base",
+                format(".*Materialized view %s.test_customer_mv_no_direct_partition_mapped must have at least one partition directly defined by a base table partition.", getSession().getSchema().get()));
+
+        // Test nested
+        assertQueryFails(
+                "CREATE MATERIALIZED VIEW test_customer_nested_mv WITH (partitioned_by = ARRAY['nationkey']) AS SELECT name, nationkey FROM test_customer_mv",
+                format(".*CreateMaterializedView on a materialized view %s.%s.test_customer_mv is not supported.", getSession().getCatalog().get(), getSession().getSchema().get()));
+
+        // Test query shape
+        assertUpdate("CREATE MATERIALIZED VIEW test_customer_agg_mv WITH (partitioned_by = ARRAY['nationkey']) AS SELECT COUNT(DISTINCT name) AS num, nationkey FROM (SELECT name, nationkey FROM test_customer_base GROUP BY 1, 2) a GROUP BY nationkey");
+        assertUpdate("CREATE MATERIALIZED VIEW test_customer_union_mv WITH (partitioned_by = ARRAY['nationkey']) AS SELECT name, nationkey FROM ( SELECT name, nationkey FROM test_customer_base WHERE nationkey = 1 UNION ALL SELECT name, nationkey FROM test_customer_base WHERE nationkey = 2)");
+        assertUpdate("CREATE MATERIALIZED VIEW test_customer_order_join_mv WITH (partitioned_by = ARRAY['orderstatus', 'nationkey']) AS SELECT orders.totalprice, orders.orderstatus, customer.nationkey FROM test_customer_base customer JOIN test_orders_base orders ON orders.custkey = customer.custkey");
+        assertQueryFails(
+                "CREATE MATERIALIZED VIEW test_customer_order_join_mv_no_base_partition_mapped WITH (partitioned_by = ARRAY['custkey']) AS SELECT orders.totalprice, customer.nationkey, customer.custkey FROM test_customer_base customer JOIN test_orders_base orders ON orders.custkey = customer.custkey",
+                format(".*Materialized view %s.test_customer_order_join_mv_no_base_partition_mapped must have at least one partition directly defined by a base table partition.", getSession().getSchema().get()));
+        assertQueryFails(
+                "CREATE MATERIALIZED VIEW test_customer_order_join_mv_no_base_partition_mapped WITH (partitioned_by = ARRAY['nation_order']) AS SELECT orders.totalprice, CONCAT(CAST(customer.nationkey AS VARCHAR), orders.orderstatus) AS nation_order FROM test_customer_base customer JOIN test_orders_base orders ON orders.custkey = customer.custkey",
+                format(".*Materialized view %s.test_customer_order_join_mv_no_base_partition_mapped must have at least one partition directly defined by a base table partition.", getSession().getSchema().get()));
+    }
```

Comment:
add `assertUpdate("DROP TABLE test_customer_base");` and the same for `test_orders_base`. We wanna clean up the created table after unit test is done.